packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -350,16 +350,19 @@ class InspectorController extends DisposableController
   }
 
   @override
-  Future<void> onForceRefresh() {
+  Future<void> onForceRefresh() async {
     assert(!_disposed);
     if (!visibleToUser || _disposed) {
       return Future.value();
     }
-    recomputeTreeRoot(null, null, false);
+    // TODO(dantup): Is it safe to add this await here?
```

Comment:
Not sure - that would be a question for @jacob314

Reply:
it is fine to add an await here. It is not ok to incrementally fetch the tree one level at a time. Fetching the tree needs to be more of a batch operation to avoid excessive round trips over the VM Service.

Reply:
This is now using `hasDiagnosticsValue` on the controller, which checks a local map that has these.